src/ccalltest.c

```diff
@@ -393,16 +397,82 @@ JL_DLLEXPORT void *test_echo_p(void *p) {
 
 #include <xmmintrin.h>
 
-JL_DLLEXPORT __m128i test_m128i(__m128i a, __m128i b, __m128i c, __m128i d ) {
+JL_DLLEXPORT __m128i test_m128i(__m128i a, __m128i b, __m128i c, __m128i d )
+{
     // 64-bit x86 has only level 2 SSE, which does not have a <4 x int32> multiplication,
     // so we use floating-point instead, and assume caller knows about the hack.
     return _mm_add_epi32(a,
                          _mm_cvtps_epi32(_mm_mul_ps(_mm_cvtepi32_ps(b),
                                                     _mm_cvtepi32_ps(_mm_sub_epi32(c,d)))));
 }
 
-JL_DLLEXPORT __m128 test_m128(__m128 a, __m128 b, __m128 c, __m128 d ) {
+JL_DLLEXPORT __m128 test_m128(__m128 a, __m128 b, __m128 c, __m128 d )
+{
     return _mm_add_ps(a, _mm_mul_ps(b, _mm_sub_ps(c, d)));
 }
 
 #endif
+
+#ifdef _CPU_AARCH64_
```

Comment:
`__int128`, `__fp16` and the vector types used below are AArch64 extensions (especially when passed as arguments or return values). I'm pretty sure at least `__fp16` is not allowed as return value elsewhere (even when similar types is defined in C).